packages/modal-ui-js/src/lib/render-modal.ts

```diff
@@ -108,9 +112,30 @@ export async function connectToWallet(module: ModuleState<Wallet>) {
       }
     }
 
+    if (wallet.type === "bridge") {
+      const subscription = modalState.selector.on("uriChanged", ({ uri }) => {
+        renderScanQRCode(module, {
+          uri,
+          handleOpenDefaultModal: () => {
+            connectToWallet(module, true);
+          },
+        });
+      });
+
+      await wallet.signIn({
+        contractId: modalState.options.contractId,
+        methodNames: modalState.options.methodNames,
+        qrCodeModal,
+      });
+
+      subscription.remove();
+      return;
```

Comment:
We need to close the Modal after successfully connecting by adding this before the return.

```ts
modalState.container.children[0].classList.remove("open");
```